src/main/java/teammates/ui/webapi/FeedbackSessionPublishedEmailWorkerAction.java

```diff
@@ -19,20 +20,42 @@ public JsonResult execute() {
         String feedbackSessionName = getNonNullRequestParamValue(ParamsNames.FEEDBACK_SESSION_NAME);
         String courseId = getNonNullRequestParamValue(ParamsNames.COURSE_ID);
 
-        FeedbackSessionAttributes session = logic.getFeedbackSession(feedbackSessionName, courseId);
+        if (!isCourseMigrated(courseId)) {
+            FeedbackSessionAttributes session = logic.getFeedbackSession(feedbackSessionName, courseId);
+            if (session == null) {
+                log.severe("Feedback session object for feedback session name: " + feedbackSessionName
+                        + " for course: " + courseId + " could not be fetched.");
+                return new JsonResult("Failure");
+            }
+            List<EmailWrapper> emailsToBeSent = emailGenerator.generateFeedbackSessionPublishedEmails(session);
+            try {
+                taskQueuer.scheduleEmailsForSending(emailsToBeSent);
+                logic.updateFeedbackSession(
+                        FeedbackSessionAttributes
+                                .updateOptionsBuilder(session.getFeedbackSessionName(), session.getCourseId())
+                                .withSentPublishedEmail(true)
+                                .build());
+            } catch (Exception e) {
+                log.severe("Unexpected error", e);
+            }
+            return new JsonResult("Successful");
+        }
+
+        FeedbackSession session = sqlLogic.getFeedbackSession(feedbackSessionName, courseId);
         if (session == null) {
             log.severe("Feedback session object for feedback session name: " + feedbackSessionName
-                       + " for course: " + courseId + " could not be fetched.");
+                    + " for course: " + courseId + " could not be fetched.");
             return new JsonResult("Failure");
         }
-        List<EmailWrapper> emailsToBeSent = emailGenerator.generateFeedbackSessionPublishedEmails(session);
+        List<EmailWrapper> emailsToBeSent = sqlEmailGenerator.generateFeedbackSessionPublishedEmails(session);
         try {
             taskQueuer.scheduleEmailsForSending(emailsToBeSent);
-            logic.updateFeedbackSession(
-                    FeedbackSessionAttributes
-                            .updateOptionsBuilder(session.getFeedbackSessionName(), session.getCourseId())
-                            .withSentPublishedEmail(true)
-                            .build());
+
+            sqlLogic.updateFeedbackSession(session.getId(), session.getInstructions(), session.getStartTime(),
+                    session.getEndTime(), session.getSessionVisibleFromTime(), session.getResultsVisibleFromTime(),
+                    session.getGracePeriod(), session.isOpeningEmailEnabled(), session.isClosingEmailEnabled(),
+                    session.isPublishedEmailEnabled(), true, session.getDeadlineExtensions(),
+                    session.getFeedbackQuestions());
```

Comment:
This method is not required.

```
session.setPublishedEmailSent(true);
```

should be enough. session will be updated when the session is flushed.
```